internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -191,3 +181,62 @@ func (c *CodePipeline) getPipelineName(resourceArn string) (string, error) {
 
 	return parsedArn.Resource, nil
 }
+
+// GetPipelineStatus retrieves status information from a given pipeline.
+func (c *CodePipeline) GetPipelineState(name string) (*PipelineState, error) {
+	input := &cp.GetPipelineStateInput{
+		Name: aws.String(name),
+	}
+	resp, err := c.client.GetPipelineState(input)
+
+	if err != nil {
+		return nil, fmt.Errorf("get pipeline state %s: %w", name, err)
+	}
+	var stageStates []*StageState
+	for _, stage := range resp.StageStates {
+		var stageName string
+		if stage.StageName != nil {
+			stageName = aws.StringValue(stage.StageName)
+		}
+		var transition string
+		if stage.InboundTransitionState != nil {
+			transition = "DISABLED"
+			if *stage.InboundTransitionState.Enabled == true {
+				transition = "ENABLED"
+			}
+		}
+		var status string
+		for _, actionState := range stage.ActionStates {
+			if actionState.LatestExecution != nil {
+				status = aws.StringValue(actionState.LatestExecution.Status)
+			}
+		}
+		stageStates = append(stageStates, &StageState{
+			StageName:  stageName,
+			Status:     status,
+			Transition: transition,
+		})
+	}
+	return &PipelineState{
+		PipelineName: aws.StringValue(resp.PipelineName),
+		StageStates:  stageStates,
+		UpdatedAt:    resp.Updated,
+	}, nil
+}
+
+// HumanString returns the stringified PipelineState struct with human readable format.
+// Example output:
+//   DeployTo-test	Deploy	Cloudformation	stackname: dinder-test-test
+func (s *StageState) HumanString() string {
+	const empty = "  -"
+	switch {
+	case s.Status == "" && s.Transition == "":
+		return fmt.Sprintf("  %s\t%s\t%s\n", s.StageName, empty, empty)
+	case s.Status == "":
+		return fmt.Sprintf("  %s\t%s\t%s\n", s.StageName, empty, s.Transition)
+	case s.Transition == "":
+		return fmt.Sprintf("  %s\t%s\t%s\n", s.StageName, s.Status, empty)
+	default:
+		return fmt.Sprintf("  %s\t%s\t%s\n", s.StageName, s.Status, s.Transition)
+	}
```

Comment:
You may be able to simplify this:

```suggestion
status := s.Status
transition := s.Transition
if status == "" {
  status = empty
}
if transition == "" {
  transition = empty
}
return fmt.Sprintf("  %s\t%s\t%s\n", s.StageName, status, transition) 
```